xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/displayer_email.vm

```diff
@@ -4,15 +4,8 @@
   <input type="text" id="$!id" name="$!id" value="$!escapetool.xml($value)" class="email"
     #if ($disabled)disabled="disabled"#end />
 #elseif ($type == 'view' || $type == 'rendered')
-  #set ($mailConfigDoc = $xwiki.getDocument('Mail.MailConfig'))
-  #set ($generalMailConfigObject = $mailConfigDoc.getObject('Mail.GeneralMailConfigClass'))
-  #set ($obfuscate = $generalMailConfigObject.getValue('obfuscate'))
-  #if ("$!obfuscate" == '')
-    ## We handle backward compatibility by also looking in the XWikiPreferences xobject
-    #set ($obfuscate = $xwiki.getXWikiPreferenceAsInt('obfuscateEmailAddresses', 0))
-  #end
-  #if ($obfuscate == 1)
-$!escapetool.xml($value.replaceAll('^(.).*@', '$1...@'))##
+  #if ($services.mailsender.generalConfiguration.obfuscateEmails)
+$!escapetool.xml($services.mailsender.obfuscateForDisplay($value))##
```

Comment:
Already done in https://github.com/xwiki/xwiki-platform/blob/3ac126fc2f6993b2f848a488a6cf62484d3e06c4/xwiki-platform-core/xwiki-platform-mail/xwiki-platform-mail-general/src/main/resources/templates/displayer_email.vm#L27-L28